### PR TITLE
Camera disabler - Uplink bug detector

### PR DESCRIPTION
**Pull request by eplgr**
The syndicate has managed to further improve their tools, and today they added new features to their Bug Detector:
# Github changelog
- The syndicate bug detector now has a camera disabler built-in feature:
  - Activating it will disable cameras for 90 seconds:
    - Cameras in a 5 tile radius will be disabled for sure
    - Cameras elsewhere have a 25% to be disabled, each one
    - Only works in your current Z-level.
  - 5 minutes cooldown.

This basically uses the effect that EMPs do to the cameras, but without an EMP.

# Explanation & thoughts
I feel like ss13 needs more tools that allow you to be sneaky, especially against an AI that pays attention. This seems reasonable, especially because after 90 seconds the cameras go back to normal again.
At first I thought of this item as a stand-alone thing, I even wanted to make it similar to how the powersink works, but after talking to some players I realized there was some potential to add it as a feature to an underused uplink item, and this disguised multitool felt perfect for the job.

# In-game changelog
:cl:
 * rscadd: Uplink bug detector: (new!) Activating it will disable cameras nearby, plus the ones far away randomly, for some time.
 * tweak: Uplink bug detector: changed price from 3 to 5 telecrystals.

### CODE REVIEW

File: code/datums/uplink_item.dm [hunk 1]
```diff
@@ -576,8 +576,8 @@ var/list/uplink_items = list()
 	cost = 1
 
 /datum/uplink_item/device_tools/bugdetector
-	name = "Bug Detector"
-	desc = "A functional multitool that can detect certain surveillance devices. Its screen changes color if the AI or a pAI can see you, or if a tape recorder or voice analyzer is nearby. Conspicuous if currently detecting something. Examine it to see everything it detects. "
+	name = "Bug Detector w/Camera Disabler built-in"
```
Comment (ghost) [inline code/datums/uplink_item.dm]: How about just "Bug Detector & Camera Disabler"?

Comment (eplgr) [reply code/datums/uplink_item.dm]: Doesn't that sound like it's a bundle/it's more than just 1 item? It crossed my mind and that's the reason I didn't choose that name. I wouldn't mind changing it though.
Also, I'm not really into the "camera disabler" name but I couldn't think of anything better.

Comment (Shadowmech88) [reply code/datums/uplink_item.dm]: "Combination Bug Detector and Camera Disabler"

**Comment by Ratvar** [comment]
Buy more camera bugs to kill AI every five minutes

**Comment by eplgr** [comment]
Changed TC price from 3 to 5. Yes, you can blind the AI for ~1minute, if you spend **all** your telecrystals.

**Comment by gbasood** [comment]
code is sound at least

**Comment by Aurx** [comment]
Interesting.  
A traitor item that causes security to randomly search everyone and steal all the multitools.

**Comment by ghost** [comment]
does it affect silicon and helmet cams?

**Comment by Shadowmech88** [comment]
Wouldn't it make the most sense for the radius to be 7 tiles, so that you can be certain you've disabled a camera if you can see it/it can see you?

**Comment by eplgr** [comment]
@HarseTheef It affects all cameras that the AI can see through I believe, it uses a global camera list which was made especifically for that.

@Shadowmech88 I guess you are right.

**Pull request merged**